app/controllers/users_controller.rb

```diff
@@ -96,13 +96,23 @@ def create
     @user.password = temp_pass
     @user.password_confirmation = temp_pass
     @user.skip_confirmation!
-
-    if @user.save
+    save_worked = false
+    begin
+      save_worked = @user.save
+    rescue => error
+      error_message = error.message
+      if error_message.include? "Duplicate entry" and error_message.include? "@"
+        flash[:error] = "Email #{@user.email} already in use"
+      else
+        flash[:error] = error_message
```

Comment:
I don't think it's good to surface raw MySQL errors to the user. Instead here just set `save_worked = False` so that the "Could not create user" message will be shown.